### DISCUSSION TITLE
Texture Atlas example does not compile, is confusing and undocumented.

### DISCUSSION THREAD

**Discussion by RylanYancey** [bevyengine/bevy#5716] [Q&A]
Bevy's Texture Atlas example here: https://github.com/bevyengine/bevy/blob/main/examples/2d/texture_atlas.rs

This does not compile on my system, with the error:
```rs
error: cannot find derive macro `Resource` in this scope
  --> src\main.rs:24:10
   |
24 | #[derive(Resource, Default)]
   |          ^^^^^^^^
   |
note: `Resource` is imported here, but it is only a trait, without a derive macro
  --> src\main.rs:4:42
   |
4  | use bevy::{asset::LoadState, prelude::*, ecs::system::Resource};
   |     
```

Not only that, but it is hard to figure out what they are trying to do. Some comments would go a long way, I don't think this is acceptable. 

The top comments mention that the goal is to create a texture atlas from a folder full of sprites, and that is understandable, but does not explain how it does this.  It also does not show how to use the texture atlas once it has been created. I can't find any guides on how to get uvs' from the atlas, for instance, and certainly not here. It lacks detail and explanation.

Thanks for your time.

**Comment by rparrett** [comment]
It looks like you are running example code from bevy's main development branch, but specifying bevy 0.8 or another version in your `Cargo.toml`.

If you're looking for examples for the latest bevy release, you can use the `latest` tag on github. 

Examples are updated in the repo as development happens, so examples from the main branch may include code that is incompatible with the latest release.

https://github.com/bevyengine/bevy/blob/latest/examples/2d/texture_atlas.rs

**Reply by DJMcNab** [reply]
Indeed. It is recommended that you read the warning at https://github.com/bevyengine/bevy/tree/main/examples#examples

**Reply by RylanYancey** [reply]
okay, that brings me to https://github.com/bevyengine/bevy/blob/v0.8.0/examples/2d/texture_atlas.rs.

This does compile. I didn't see that warning.

I don't like how little comments you guys have. Maybe if I can figure this stuff out I'll try and write some.
